aptos-move/move-examples/token_v2/sources/collection.move

```diff
@@ -0,0 +1,450 @@
+/// * Consider supporting changing the aspects of supply
+/// * Add aggregator support when added to framework
+/// * Update ObjectId to be an acceptable param to move
+module token_v2::collection {
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::String;
+
+    use aptos_framework::object::{Self, Object, generate_transfer_ref, object_address, disable_ungated_transfer, object_from_constructor_ref};
+    use aptos_std::smart_table::SmartTable;
+    use aptos_std::smart_table;
+    use token_v2::common::{Royalty, new_royalty, exists_royalty, remove_royalty};
+    use token_v2::refs::{Refs, new_refs_from_constructor_ref, generate_object_from_refs, get_delete_from_refs, address_of_refs};
+    #[test_only]
+    use std::string;
+    #[test_only]
+    use std::signer::address_of;
+    use token_v2::common;
+
+    friend token_v2::token;
+
+    /// The collections supply is at its maximum amount.
+    const EEXCEEDS_MAX_SUPPLY: u64 = 1;
+    /// The error of collection refs.
+    const ECOLLECTION_REFS: u64 = 2;
+    /// The provided signer is not the creator.
+    const ENOT_CREATOR: u64 = 3;
+    /// Attempted to mutate an immutable field.
+    const EFIELD_NOT_MUTABLE: u64 = 4;
+    /// The error of collection resource.
+    const ECOLLECTION: u64 = 5;
+    /// The error of collection index.
+    const ECOLLECTION_INDEX: u64 = 6;
+    /// The error of collection name.
+    const ECOLLECTION_NAME: u64 = 7;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Represents the common fields for a collection.
+    struct Collection has key {
+        /// The creator of this collection.
+        creator: address,
+        /// A brief description of the collection.
+        description: String,
+        /// Determines which fields are mutable.
+        mutability_config: MutabilityConfig,
+        /// An optional categorization of similar token.
+        name: String,
+        /// The Uniform Resource Identifier (uri) pointing to the JSON file stored in off-chain
+        /// storage; the URL length will likely need a maximum any suggestions?
+        uri: String,
+    }
+
+    /// This config specifies which fields in the TokenData are mutable
+    struct MutabilityConfig has copy, drop, store {
+        description: bool,
+        uri: bool,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Aggregable supply tracker, this is can be used for maximum parallel minting but only for
+    /// for uncapped mints. Currently disabled until this library is in the framework.
+    struct AggregableSupply has key {}
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Fixed supply tracker, this is useful for ensuring that a limited number of tokens are minted.
+    struct FixedSupply has key {
+        current_supply: u64,
+        max_supply: u64,
+    }
+
+    struct CollectionIndex has key {
```

Comment:
So we'll store the refs for all created collections in the creator account. Is this to give creator access to the collection object's signer?